src/components/btnExpand/BtnExpand.js

```diff
@@ -0,0 +1,68 @@
+import { FaChevronDown, FaChevronUp } from "react-icons/fa";
+import { useState, useEffect, useCallback } from "react";
+import "./style.css";
+
+const BtnExpand = (props) => {
+  const [selectedIcon, setSelectedIcon] = useState(
+    <FaChevronDown className="expandIcon" />
+  );
+  const [isComponentVisible, setComponentVisibility] = useState(false);
+  const [progress, setProgress] = useState(0);
+
+  const animateProgress = useCallback(() => {
+    let progressInterval = setInterval(() => {
+      setProgress((prevProgress) =>
+        prevProgress >= props.learnt ? props.learnt : prevProgress + 1
+      );
+    }, 30);
+
+    setTimeout(() => {
+      clearInterval(progressInterval);
+    }, 3000);
+  }, [props.learnt, setProgress]);
+
+  useEffect(() => {
+    if (isComponentVisible) {
+      animateProgress();
+    }
+  }, [isComponentVisible, animateProgress]);
+
+  const toggleIcon = () => {
+    setSelectedIcon((prevIcon) =>
+      prevIcon.type === FaChevronDown ? (
+        <FaChevronUp class="expandIcon" />
+      ) : (
+        <FaChevronDown class="expandIcon" />
+      )
+    );
+
+    setComponentVisibility((isComponentVisible) =>
+      isComponentVisible === true ? false : true
```

Comment:
```suggestion
      !isComponentVisible
```
Согласитесь, так короче?
Скорее всего, eslint вам бы такое подсказал. давайте его подключим.